src/ResourceManager/Consumption/Commands.Consumption/Cmdlets/GetAzureRmConsumptionPriceSheet.cs

```diff
@@ -0,0 +1,120 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Consumption.Common;
+using Microsoft.Azure.Commands.Consumption.Models;
+using Microsoft.Azure.Management.Consumption;
+using Microsoft.Azure.Management.Consumption.Models;
+
+namespace Microsoft.Azure.Commands.Consumption.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmConsumptionPriceSheet"), OutputType(typeof(PSPriceSheet))]
+    public class GetAzureRmConsumptionPriceSheet : AzureConsumptionCmdletBase
+    {
+        private const int MaxNumberToFetch = 1000;
+
+        [Parameter(Mandatory = false, HelpMessage = "Name of a specific billing period to get the price sheets that associate with.")]
+        [ValidateNotNullOrEmpty]
+        public string BillingPeriodName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Expand the price sheets based on MeterDetails.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("MeterDetails")]
```

Comment:
Is this going to have more values eventually? Because otherwise it seems like this parameter should just be a switch parameter "ExpandMeterDetails"

Reply:
We don't have plans to add more values yet. So make it `ExpandMeterDetail`